apps/web/src/components/Home/SeeThroughLens.tsx

```diff
@@ -0,0 +1,179 @@
+import UserProfile from '@components/Shared/UserProfile';
+import { Input } from '@components/UI/Input';
+import { Spinner } from '@components/UI/Spinner';
+import { Menu, Transition } from '@headlessui/react';
+import { XIcon } from '@heroicons/react/outline';
+import { ChevronDownIcon } from '@heroicons/react/solid';
+import getAvatar from '@lib/getAvatar';
+import { Leafwatch } from '@lib/leafwatch';
+import clsx from 'clsx';
+import type { FeedItem, Profile } from 'lens';
+import {
+  CustomFiltersTypes,
+  SearchRequestTypes,
+  useSearchProfilesLazyQuery,
+  useTimelineLazyQuery
+} from 'lens';
+import type { ChangeEvent, FC } from 'react';
+import React, { Fragment, useState } from 'react';
+import { useAppStore } from 'src/store/app';
+import { useTimelineStore } from 'src/store/timeline';
+import { SEARCH } from 'src/tracking';
+
+const SeeThroughLens: FC = () => {
+  const currentProfile = useAppStore((state) => state.currentProfile);
+  const seeThroughProfile = useTimelineStore((state) => state.seeThroughProfile);
+  const setSeeThroughProfile = useTimelineStore((state) => state.setSeeThroughProfile);
+
+  const [recommendedProfilesToSeeThrough, setRecommendedProfilesToSeeThrough] = useState<Profile[]>([]);
+  const [searchText, setSearchText] = useState('');
+
+  const setRecommendedProfiles = (feedItems: FeedItem[]) => {
+    let uniqueProfileIds: string[] = [];
+    let profiles: Profile[] = [];
+    for (const feedItem of feedItems) {
+      const profileId = feedItem.root?.profile?.id;
+      if (
+        !uniqueProfileIds.includes(profileId) &&
+        profileId !== seeThroughProfile?.id &&
+        profileId !== currentProfile?.id
+      ) {
+        profiles.push(feedItem.root?.profile as Profile);
+        uniqueProfileIds.push(profileId);
+      }
+    }
+    setRecommendedProfilesToSeeThrough(profiles?.slice(0, 5));
+  };
+
+  const profile = seeThroughProfile ?? currentProfile;
+  const request = { profileId: profile?.id, limit: 50 };
+
+  const [searchUsers, { data: searchUsersData, loading: searchUsersLoading }] = useSearchProfilesLazyQuery();
+
+  const [fetchRecommendedProfiles, { loading, error }] = useTimelineLazyQuery({
```

Comment:
Can we make a new query to get only profiles?

Reply:
to get timeline users?

Reply:
yes